cmd/kubelet/app/server.go

```diff
@@ -638,7 +646,12 @@ func RunKubelet(kcfg *KubeletConfig, builder KubeletBuilder) error {
 	eventBroadcaster.StartLogging(glog.V(3).Infof)
 	if kcfg.KubeClient != nil {
 		glog.V(4).Infof("Sending events to api server.")
-		eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""))
+		if kcfg.EventRecordQPS == 0.0 {
+			eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""), nil)
+		} else {
+			eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""), util.NewTokenBucketRateLimiter(kcfg.EventRecordQPS, kcfg.EventBurst))
```

Comment:
@lavalamp Thanks a lot for your idea. Totally agree with you. I think this will make event recorder behavior more clear. Will update this pr soon.